quickwit/quickwit-indexing/src/source/ingest_api_source.rs

```diff
@@ -84,13 +85,9 @@ impl IngestApiSource {
         };
         ingest_api_service.ask_for_res(create_queue_req).await?;
 
-        let previous_offset = if let Some(Position::Offset(offset_str)) =
-            checkpoint.position_for_partition(&partition_id)
-        {
-            Some(offset_str.parse::<u64>()?)
-        } else {
-            None
-        };
+        let previous_offset = checkpoint
+            .position_for_partition(&partition_id)
+            .map(|position| position.as_u64().expect("offset should be stored as u64"));
```

Comment:
I think we should log an error and index from the beginning.

Reply:
Explanation:

The idea here is that a wrong checkpoint will stale prod.
Generally speaking, I am uncomfortable with expect that are justified by something that is "distant":
- distant in code
- time
- hardware etc.

(I say that but a corrupted tantivy index panics)

Reply:
I opened #4025. You can ignore this.